Shopify/cli
# Remove the support for declarative webhooks

### WHY are these changes introduced?
Finally, the `declarative webhooks` project won't be shipped for Winter Editions'24. Pushing webhooks modules with declarative subscriptions should not be supported.
However `api_version` should be supported in the server side, otherwise `versioned app config` won't work



### WHAT is this pull request doing?
- Remove support for `subscriptions` from the `webhooks` spec schema


### How to test your changes?
- Create a remote app using the `main` branch `p shopify app config link --path /PATH/TO/YOUR/APP`
- Enable the `versioned_app_config` beta
- Add some `webhooks suscriptions` to the `toml`
- Run `p shopify app deploy --path /PATH/TO/YOUR/APP`
- Checkout the cli branch `remove-declarative-webhooks-support`
- Run `p shopify app deploy --path /PATH/TO/YOUR/APP`. You should see an error banner
- Remove the `webhooks suscriptions` to the `toml`
- Run `p shopify app link --path /PATH/TO/YOUR/APP` selecting the previous app that contains `declarative webhooks`
- Run `p shopify app deploy --path /PATH/TO/YOUR/APP`
- 

### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
